Sj example database and topcat queries

stephjuneau opened the pull request:
- Added new notebook for public database access in 01_getting_started.
- Fixed affiliation in nb 04

Update README files for:
- tutorial top level (small fixes)
- 01_getting_started: more substantial content including tutorial Overview of all notebooks and example TAP usage with astroquery
- database/ folder to clarify NERSC vs. public databases

stephjuneau committed:
new getting_start db notebook and updated readmes

stephjuneau committed:
fixed affiliations nb 04, 06

sbailey commented:
Overall looks good.  @stephjuneau and I sorted out an issue with python 3.12 string support, and an edge case when the randomly selected spectra is flagged as being entirely bad.  She'll update after a meeting.

stephjuneau committed:
fixed str fmt and removed spectype cut

stephjuneau commented:
@sbailey -- I made the following changes:
- added "r" in front of string labels
- added ZWARN on all panels with an explainer below (and link to bit definitions)
- removed SPECTYPE from query as it only affects 2 records (now finds 782 instead of 780 -- YEP the same two that you randomly found... out of 23 millions)

stephjuneau committed:
fixed r string

sbailey reviewed (approved):
Looks good, thanks.  I confirm that I can run the notebook without errors from my laptop without requiring a NERSC or NOIRLab account or any local DESI data.

I suggest that a future update add additional context at the start about the pros and cons of using the database vs. files.  For right now, I will merge this so that we have it in place for the public release in 27 minutes.
